### ISSUE TITLE
formatter breaks code

### ISSUE THREAD

**Issue by olehmisar** [noir-lang/noir#6333]
### Aim

`nargo fmt` changes code behaviour 
https://github.com/olehmisar/nodash/blob/e15265ec23be8686d89a20b60861c7f588ea5a47/src/array.nr#L44

```rs
let bytes_padded = bytes.pad_end::<(N / 31 + 1) * 31>(0);
```

It converts it to:
```rs
let bytes_padded = bytes.pad_end::<(N / 31 + 1) * 31>(N / 31 + 1) * 31>( 0);
```

### Expected Behavior

Does not change code, but only formats it

### Bug

^^

### To Reproduce

1. git clone https://github.com/olehmisar/nodash.git
2. cd nodash
3. nargo fmt

### Workaround

None

### Project Impact

Nice-to-have

### Nargo Version

nargo 0.34.0 and nargo 0.35.0

### Would you like to submit a PR for this Issue?

None

**Comment by TomAFrench** [comment]
@asterite Can you take a look at this?

**Comment by asterite** [comment]
Sure! I just took a look, it seems to be a bug in the old formatter. In the new formatter it works fine 🙌 

Fixed by #6300
